CRM/Core/Form/RecurringEntity.php

```diff
@@ -399,6 +399,7 @@ public static function postProcess($params = array(), $type, $linkedEntities = a
                 'is_active' => 1,
               );
               $excludeDateList[] = $optionGroupValue['value'];
+              // Something is wrong here. The returned BAO object is not being used.
               CRM_Core_BAO_OptionValue::create($optionGroupValue);
```

Comment:
Right & I was discussing that sql query with  @totten  today - with a view to whether we could do it less often - this is his PR on it https://github.com/civicrm/civicrm-core/pull/7215 - but after looking at your comments & doing a little reading I'm thinking the query might need to go deeper...

http://dev.mysql.com/doc/refman/5.7/en/charset-connection.html 
http://akrabat.com/utf8-php-and-mysql/

What I take from the above is that currently we are calling set UTF Names whenever a DAO object is initialised. (Potentially a BAO might override the construct function and override that). The DAO seems to be initialised in the executeQuery and the singleValueQuery & I haven't quite figured out how it could be being bypassed yet. I'm wondering if it is happening at the php or sql level.

OTOH we aspire to that SET names UTF running less frequently. Currently because it runs at a higher level than the connection it runs regardless of whether the connection is already initiated.

I wonder if we can pass it into options here in some way? https://github.com/civicrm/civicrm-packages/blob/master/DB.php#L525 - the second link above has a few variants.

NB - although in theory we shouldn't hack packages we .... sometimes do....